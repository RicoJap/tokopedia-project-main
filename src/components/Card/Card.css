.Card-wrapper {
    margin-bottom: 1em;
    cursor: pointer;
}