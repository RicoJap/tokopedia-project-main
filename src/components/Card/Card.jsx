import React from "react";
import MuiCard from "@material-ui/core/Card";
import MuiCardActions from "@material-ui/core/CardActions";
import MuiCardContent from "@material-ui/core/CardContent";
import Typography from "../Typography/Typography";
import Divider from "@material-ui/core/Divider";
import { makeStyles } from "@material-ui/core/styles";

const useCardStyles = makeStyles({
  root: {
    marginBottom: '1em',
    cursor: 'pointer'
  },
});

/**
 * Card Component
 *
 * @since  15/11/2020
 * @access (public)
 *
 * @type     Functional Component
 * @param {String}  label  Card label
 * @param {NUMBER}  labelVariant  Label HTML text tag variant e.g. h1, h2, h3, p
 * @param {ReactNode}  children  Children DOM node of the card
 * @param {Object}  styles  UYser defined styles
 * @param {Boolean}  separateContentAndButtons  Add a divider to separate content and buttons
 *
 */

const Card = ({
  label,
  labelVariant = "h6",
  subLabel,
  children,
  styles = {},
  separateContentAndButtons = false,
  ...attributes
}) => {

	const cardActionsChildren = children;
	const cardStyles = useCardStyles();

  return (
    <MuiCard classes={cardStyles} {...attributes}>
      <MuiCardContent style={styles.cardContent}>
        <Typography label={label} variant={labelVariant} data-testid="card-label" />
        {!!subLabel && <Typography label={subLabel} variant="subtitle2" color="textSecondary" data-testid="card-sub-label" />}
        {separateContentAndButtons && <Divider style={styles.divider} />}
        {!!cardActionsChildren ? (
          <MuiCardActions style={styles.cardActions}>
            {cardActionsChildren}
          </MuiCardActions>
        ) : null}
      </MuiCardContent>
    </MuiCard>
  );
};

export default Card;
