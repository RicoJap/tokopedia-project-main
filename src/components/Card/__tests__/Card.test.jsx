import "@testing-library/jest-dom";
import React from "react";
import renderer from "react-test-renderer";

import { fireEvent, render, screen, waitFor } from "@testing-library/react";

import Card from "../Card";

const MockCard = () => {
  return <Card label={"Test Card"} />;
};

describe("Snapshot testing", () => {
  const tree = renderer.create(<MockCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Card component renders correctly", () => {
  test("Card component is present in the document", () => {
    render(<MockCard />);
    expect(screen.getByTestId("card-label")).toBeInTheDocument();
  });
});
