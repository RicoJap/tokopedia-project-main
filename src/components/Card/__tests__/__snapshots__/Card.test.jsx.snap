// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="card-label"
    >
      Test Card
    </h6>
  </div>
</div>
`;
